src/components/AttachmentModal.js

```diff
@@ -42,28 +45,91 @@ const propTypes = {
 };
 
 const defaultProps = {
-    title: '',
-    sourceURL: null,
-    onConfirm: null,
-    isAuthTokenRequired: false,
+    sortedReportActions: [],
+    visibleAttachment: {
+        sourceURL: null,
+        isAttachment: false,
+        isModalOpen: false,
+        file: {name: ''},
+    },
 };
 
 class AttachmentModal extends Component {
     constructor(props) {
         super(props);
 
+        this.handleKeyPress = this.handleKeyPress.bind(this);
+        this.getNextAttachment = this.getNextAttachment.bind(this);
+        this.getAttachmentSource = this.getAttachmentSource.bind(this);
+
         this.state = {
-            isModalOpen: false,
-            file: null,
-            sourceURL: props.sourceURL,
+            title: 'Attachment',
         };
     }
 
+    componentDidUpdate(prevProps) {
+        const {visibleAttachment: {isModalOpen}} = this.props;
+        if (!prevProps.visibleAttachment.isModalOpen && isModalOpen) {
+            document.addEventListener('keydown', this.handleKeyPress, false);
+        } else {
+            document.removeEventListener('keydown', this.handleKeyPress, false);
+        }
+    }
+
+    // Using regex to get the src attribute from the html img tag
+    getAttachmentSource(htmlString) {
+        const imgRegx = /<img.*?data-expensify-source="(.*?)"[^>]+>/g;
+        const imageGroup = imgRegx.exec(htmlString);
+        return imageGroup[1];
+    }
+
+    // A function to get the next attachment based on the current attachment being show in the report.
+    getNextAttachment = (toRight) => {
+        const {sortedReportActions, visibleAttachment: {sourceURL}} = this.props;
+        const attachments = sortedReportActions.filter(sortedReportAction => sortedReportAction.action.isAttachment);
```

Comment:
We use the underscore variations of array methods for consistency.

Reply:
Using `isAttachment` is not reliable. If we want it to be we must set this when setting the report actions data here:

https://github.com/Expensify/Expensify.cash/blob/b2c03fc5a8bd0d2c5a3f67847d132af1d41702e1/src/libs/actions/Report.js#L405-L412

Reply:
I don't understand? I was using a property that was available to me via the action object.

Reply:
`isAttachment` is a property set for uploading attachments or attachments that are handled via Pusher. This does not come from the server therefore it is unreliable. e.g. here is a report action with an attachment after uploading, logging out, and logging back in.

![Screen Shot 2021-01-13 at 11 05 21 AM](https://user-images.githubusercontent.com/32969087/104510449-4f0f0200-558f-11eb-8014-0c98311ea0ed.png)

Reply:
ok